src/erlcloud_ddb2.erl

```diff
@@ -838,6 +834,65 @@ conditional_op_opt() ->
 expected_opt() ->
     {expected, <<"Expected">>, fun dynamize_expected/1}.
 
+-spec filter_expression_opt() -> opt_table_entry().
+
+filter_expression_opt() ->
+    {filter_expression, <<"FilterExpression">>, fun dynamize_expression/1}.
```

Comment:
This type spec and option entry function is added but never used elsewhere, which may cause compile warnings.
Actually, I've finished these refactors moths ago, either you could complete the refactors in `q` and `scan`, or you just delete them and I'll make a PR later.